src/Core/src/Stream/GeneratorIterableStream.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace AsyncAws\Core\Stream;
+
+use AsyncAws\Core\Exception\InvalidArgument;
+
+/**
+ * Convert an iterator into a Stream.
+ *
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ *
+ * @internal
+ */
+final class GeneratorIterableStream extends IterableStream implements ReadOnceResultStream
```

Comment:
Did we introduce this class just to add the `ReadOnceResultStream`?

Reply:
The purpose of this class is to distinguish array (rewindable) from generator (read once)

If you think it don't worth it, I can remove it.